### PR TITLE
[DNM] DOXYGEN: doxygen changes in header files 2

### PR DESCRIPTION
**Pull request by b-gent**
Edited files in the common folder:

- bufferedstream.h
- callback.h
- config-manager.h
- coroutines.h
- cosinetables.h
- dcl.h
- debug.h
- debug-channels.h

Output is available on https://b-gent.github.io/PR2488/group__common__buffstream.html

### CODE REVIEW

**Comment by b-gent** [comment]
- bufferedstream.h looking very good

Descriptions required for the following:

callback.h
-----------
I don't fully get these callbacks, I added some descriptions (please check() but the following are still missing:
- GlobalFunctionCallback(GlobalFunction cb) (line 70)
- Callback(T *object, TMethod method) (line 99)
- typedef void(T::*TMethod) (line 95)
- T *_object (line 96) 
- TMethod _method (line 97)
- CallbackBridge(T *object, TCallbackMethod method, BaseCallback<OS> *outerCallback) (line 141)

config-manager.h
--------------------
- typedef HashMap<String, Domain, IgnoreCase_Hash, IgnoreCase_EqualTo> DomainMap (line 94)
- 	bool				hasKey(const String &key) const; -> not sure how to define this 'key' and thus don't know how to document the structures that use it as an argument
	const String &		get(const String &key) const;
	void				set(const String &key, const String &value);  (lines 129-131)
- 	bool				hasKey(const String &key, const String &domName) const;
	const String &		get(const String &key, const String &domName) const;
	void				set(const String &key, const String &value, const String &domName);
	void				removeKey(const String &key, const String &domName);    (lines 151-155)
- friend class Singleton<SingletonBaseType> (line 201)
- 	void				registerDefault(const String &key, const String &value);
	void				registerDefault(const String &key, const char *value);
	void				registerDefault(const String &key, int value);
	void				registerDefault(const String &key, bool value);   (lines 170-173)
- 	bool				hasGameDomain(const String &domName) const;
	bool				hasMiscDomain(const String &domName) const; (lines 190-191)
- 	DomainMap::iterator beginGameDomains() { return _gameDomains.begin(); 
	DomainMap::iterator endGameDomains() { return _gameDomains.end();  (lines 194-195)
- void copyFrom(ConfigManager &source); (line 198)
- ALL members of the Domain class are undocumented (lines 66-91)

**Comment by sev-** [comment]
This needs rebase now.

**Comment by b-gent** [comment]
> This needs rebase now.

Rebased.

**Comment by sev-** [comment]
Good progress, merging.

File: common/callback.h [hunk 1]
```diff
@@ -69,43 +68,45 @@ template<typename T> class GlobalFunctionCallback: public BaseCallback<T> {
...
  *         &MyClass::myMethod
  *     )
+ * @endcode
  */
 template<class T, typename S = void *> class Callback: public BaseCallback<S> {
 protected:
 	typedef void(T::*TMethod)(S);
 	T *_object;
-	TMethod _method;
+		;
```
Comment (sev-) [inline common/callback.h:96]: That broke the compilation horribly, for obvious reasons.

Comment (b-gent) [reply common/callback.h:96]: To be honest, I was assuming we would still be working on this PR, at least adding missing doc in the high-priority headers. Hence the 'Do Not Merge' label in its title.

**Pull request merged**